packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -0,0 +1,395 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'constants.dart';
+import 'toggleable.dart';
+
+// Examples can assume:
+// bool _throwShotAway = false;
+// late StateSetter setState;
+
+// The relative values needed to transform a color to it's equivilant focus
+// outline color.
+const double _kCupertinoFocusColorOpacity = 0.80;
+const double _kCupertinoFocusColorBrightness = 0.69;
+const double _kCupertinoFocusColorSaturation = 0.835;
+
+/// A macOS style checkbox.
+///
+/// The checkbox itself does not maintain any state. Instead, when the state of
+/// the checkbox changes, the widget calls the [onChanged] callback. Most
+/// widgets that use a checkbox will listen for the [onChanged] callback and
+/// rebuild the checkbox with a new [value] to update the visual appearance of
+/// the checkbox.
+///
+/// The checkbox can optionally display three values - true, false, and null -
+/// if [tristate] is true. When [value] is null a dash is displayed. By default
+/// [tristate] is false and the checkbox's [value] must be true or false.
+///
+/// In the Apple Human Interface Guidelines (HIG), checkboxes are encouraged for
+/// use on macOS, but is silent about their use on iOS. If a multi-selection
+/// component is needed on iOS, the HIG encourages the developer to use switches
+/// ([CupertinoSwitch] in Flutter) instead, or to find a creative custom
+/// solution.
+///
+/// See also:
+///
+///  * [Checkbox], the Material Design equivalent.
+///  * [CupertinoSwitch], a widget with semantics similar to [CupertinoCheckbox].
+///  * [CupertinoSlider], for selecting a value in a range.
+///  * <https://developer.apple.com/design/human-interface-guidelines/components/selection-and-input/toggles/>
```

Comment:
Maybe describe what this is a link to?  (or make it into an actual Markdown link to the descriptive text?)